shared/actions/chat/inbox.js

```diff
@@ -274,20 +274,31 @@ function* untrustedInboxVisible(action: Constants.UntrustedInboxVisible): SagaGe
   }
 }
 
+const _chatInboxToProcess = []
+
 function* _chatInboxConversationSubSaga({conv}) {
-  // Wait for an idle
-  yield call(onIdlePromise, 100)
-  // TODO might be better to make this a put with an associated takeEvery
-  yield spawn(processConversation, conv)
+  _chatInboxToProcess.push(conv)
+  yield put(Creators.unboxMore())
   return EngineRpc.rpcResult()
 }
 
+function* unboxMore(): SagaGenerator<any, any> {
+  if (!_chatInboxToProcess.length) {
+    return
+  }
+
+  const conv = _chatInboxToProcess.pop()
```

Comment:
did you mean `shift`? Currently this behaves like a stack, but it sounds like you intended a queue.

Reply:
no i meant pop as i figured the most recent thing you asked for is likely what you want (aka scrolling and etc)